src/Events/Admin/Onboarding/Controller.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Controller for interfacing with TEC\Common\Onboarding.
+ *
+ * @since 7.0.0
+ */
+
+namespace TEC\Events\Admin\Onboarding;
+
+use TEC\Common\Contracts\Provider\Controller as Controller_Contract;
+use TEC\Events\Admin\Onboarding\API;
+use TEC\Events\Admin\Onboarding\Steps\Optin;
+use TEC\Events\Admin\Onboarding\Steps\Settings;
+use TEC\Events\Admin\Onboarding\Steps\Organizer;
+use TEC\Events\Admin\Onboarding\Steps\Venue;
+use TEC\Events\Admin\Onboarding\Steps\Tickets;
+use TEC\Events\Admin\Onboarding\Data;
+use TEC\Events\Admin\Onboarding\Landing_Page;
+
+/**
+ * Class Controller
+ *
+ * @since 7.0.0
+ * @package TEC\Events\Admin\Onboarding
+ */
+class Controller extends Controller_Contract {
+
+	/**
+	 * Register the provider.
+	 *
+	 * @since 7.0.0
+	 */
+	public function do_register(): void {
+		$this->add_filters();
+		$this->add_actions();
+
+		// Register the service provider itself on the container.
+		$this->container->singleton( static::class, $this );
+		$this->container->singleton( Landing_Page::class, Landing_Page::class );
+		$this->container->singleton( 'tec.events.onboarding.landing-page', Landing_Page::class );
+		$this->container->singleton( Data::class, Data::class );
```

Comment:
```suggestion
		$this->container->singleton( Landing_Page::class );
		$this->container->singleton( Data::class );
```

You can skip second param if its the same as the 1st. Do we need to still set strings to refer to implementations ?

Reply:
Nah, was more habit. accepting your suggestion.

Reply:
Because there isn't "one controller up" here - these are being registered in Tribe/Main.php where we don't have te container handy.